workflows/prognostic_c48_run/runtime/segmented_run/run.py

```diff
@@ -33,17 +33,22 @@ def run_segment(config: dict, rundir: str):
         x, y = config["namelist"]["fv_core_nml"]["layout"]
         nprocs = x * y * 6
         with open("logs.txt", "w") as f:
-            subprocess.check_call(
+            process = subprocess.Popen(
                 [
                     "mpirun",
                     "-n",
                     str(nprocs),
                     sys.executable,
+                    "-m",
+                    "mpi4py",
                     runfile.absolute().as_posix(),
                 ],
-                stdout=f,
-                stderr=f,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
             )
+            for c in iter(lambda: process.stdout.read(1), b""):
+                sys.stdout.buffer.write(c)
+                f.buffer.write(c)
```

Comment:
Was there some reason you used "buffer"?
```suggestion
                sys.stdout.write(c)
                f.write(c)
```

Reply:
No particular reason, other than that the example I was working off of used it.

Reply:
I guess the example had it because `stdout.write(), f.write()` take str args, so you can either decode then use those, or write to the buffer.